src/core/MOM_open_boundary.F90

```diff
@@ -339,14 +365,16 @@ subroutine open_boundary_config(G, US, param_file, OBC)
   integer :: l ! For looping over segments
   logical :: debug_OBC, debug, mask_outside, reentrant_x, reentrant_y
   character(len=15) :: segment_param_str ! The run-time parameter name for each segment
-  character(len=100) :: segment_str      ! The contents (rhs) for parameter "segment_param_str"
+  character(len=1024) :: segment_str      ! The contents (rhs) for parameter "segment_param_str"
   character(len=200) :: config1          ! String for OBC_USER_CONFIG
   real               :: Lscale_in, Lscale_out ! parameters controlling tracer values at the boundaries [L ~> m]
+  integer, dimension(3) :: tide_ref_date, nodal_ref_date  !< Reference date (t = 0) for tidal forcing
```

Comment:
These dates should probably be stored using the standard time_type, rather than inventing a new 3-integer type.

Reply:
Is there a cleaner way you would recommend for reading in a date from MOM_input? My current strategy is to set the date in MOM_input with
`OBC_TIDE_NODAL_REF_DATE = 1981,7,2` 
and then the subroutine `get_param` reads this in as a 3-integer array, which later gets does get converted to a `time_type`. 

I could do the conversion to `time_type` immediately in this subroutine rather than passing the 3-int array over to `initialize_obc_tides` for conversion if that would make it cleaner.